Adding summary methods

pstjohn opened the pull request:
This PR addresses #223, i.e., adding summary methods to model and metabolite objects.

The summary code is sadly quite messy, but there isn't a great way to pretty-print objects without adding yet another dependency.

I also had to change the Reaction.build_reaction_string method slightly in order to make sure it is deterministic for testing purposes.

pstjohn commented:
Give me some time to work out the quirks here, we'll also want to make sure travis installs pandas.

pstjohn commented:
Ok, this is now working largely as intended. Let me know if you have any thoughts on the implementation. Hopefully the checks all pass! (they did before I rebased...)

aebrahim commented:
I reviewed this pull request, and it's excellent. Thank you for doing this, and I'm definitely going to merge it.

If it's not too much trouble, would you mind doing the following first however?

1. Adding that the model needs to have already been solved to the docstring of model.summary
2. Running ```git commit --amend --reset-author``` and editing the commit message to describe the feature in 1-2 sentences?

After this gets merged, I'm going to add in an example of use to the end of ```documentation_builder/simulating.ipynb``` notebook. Would you mind looking it over (alternately do you want to write it)?

aebrahim commented on cobra/core/Reaction.py:
```diff
@@ -541,7 +541,8 @@ def format(number):
         reactant_bits = []
         product_bits = []
-        for the_metabolite, coefficient in iteritems(self._metabolites):
+        for the_metabolite, coefficient in sorted(
```
Good call :+1:

pstjohn commented:
No problem! Updated the docstrings, added the documentation, and rebased the commit with a more informative commit message. Glad you like it, this was definitely a feature I thought was needed :)
